addons/l10n_ec/models/account_move.py

```diff
@@ -0,0 +1,244 @@
+                    identification_code = "05"
+                elif is_passport:
+                    identification_code = "06"
+            else:
+                if is_ruc:
+                    identification_code = "20"
+                elif is_dni:
+                    identification_code = "21"
+                else:
+                    identification_code = "09"
+        return identification_code
+
+    @api.model
+    def _get_l10n_ec_documents_allowed(self, identification_code):
+        documents_allowed = self.env['l10n_latam.document.type']
+        for document_ref in _DOCUMENTS_MAPPING.get(identification_code, []):
+            document_allowed = self.env.ref('l10n_ec.%s' % document_ref, False)
+            if document_allowed:
+                documents_allowed |= document_allowed
+        return documents_allowed
+
+    def _get_l10n_ec_internal_type(self):
+        self.ensure_one()
+        internal_type = self.env.context.get("internal_type", "invoice")
+        if self.move_type in ("out_refund", "in_refund"):
+            internal_type = "credit_note"
+        if self.debit_origin_id:
+            internal_type = "debit_note"
+        return internal_type
+
+    def _get_l10n_latam_documents_domain(self):
+        if self.company_id.country_id.code != "EC":
+            return super(AccountMove, self)._get_l10n_latam_documents_domain()
+        domain = []
+        internal_type = self._get_l10n_ec_internal_type()
+        allowed_documents = self._get_l10n_ec_documents_allowed(self._get_l10n_ec_identification_type())
+        if internal_type and allowed_documents:
+            domain.append(("id", "in", allowed_documents.filtered(lambda x: x.internal_type == internal_type).ids))
+        return domain
+
+    def _get_ec_formatted_sequence(self, number=0):
+        return "%s %s-%s-%09d" % (
+            self.l10n_latam_document_type_id.doc_code_prefix,
+            self.journal_id.l10n_ec_entity,
+            self.journal_id.l10n_ec_emission,
+            number,
+        )
+
+    def _get_starting_sequence(self):
+        """If use documents then will create a new starting sequence using the document type code prefix and the
+        journal document number with a 8 padding number"""
+        if (
+            self.journal_id.l10n_latam_use_documents
+            and self.company_id.country_id.code == "EC"
+        ):
+            if self.l10n_latam_document_type_id:
+                return self._get_ec_formatted_sequence()
+        return super()._get_starting_sequence()
+
+    def _get_last_sequence_domain(self, relaxed=False):
+        l10n_latam_document_type_model = self.env['l10n_latam.document.type']
+        if self.country_code == "EC" and self.move_type in (
+            "out_invoice",
+            "out_refund",
+            "in_invoice",
+            "in_refund",
+        ):
+            where_string, param = super(AccountMove, self)._get_last_sequence_domain(
+                relaxed=True
```

Comment:
relaxed=relaxed

Reply:
I know, but I just want to avoid messing with the inheritance.  Unless you tell me that putting relaxed=relaxed will give problems.

Reply:
relaxed=relaxed make where_string return with all date pattern, because default value is False, in the last super call I give original variable value to make inheritance return default values , on this context inside if I need to specify value of super call to get correct where_string and params values